cmd/srcd/cmd/stop_test.go

```diff
@@ -0,0 +1,108 @@
+// +build integration
+
+package cmd
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"testing"
+
+	cmdtest "github.com/src-d/engine/cmd/test-utils"
+	"github.com/src-d/engine/docker"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type StopTestSuite struct {
+	cmdtest.IntegrationSuite
+	testDir string
+}
+
+func TestStopTestSuite(t *testing.T) {
+	s := StopTestSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *StopTestSuite) SetupTest() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "stop-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (s *StopTestSuite) TearDownTest() {
+	s.RunStop(context.Background())
+	os.RemoveAll(s.testDir)
+}
+
+func (s *StopTestSuite) TestInitStop() {
+	require := s.Require()
+
+	_, err := s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	_, err = s.RunSQL(context.TODO(), "SELECT 1")
+	require.NoError(err)
+
+	_, err = s.RunStop(context.TODO())
+	require.NoError(err)
+
+	s.allStopped()
+}
+
+func (s *StopTestSuite) TestStopTwice() {
+	require := s.Require()
+
+	_, err := s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	_, err = s.RunStop(context.TODO())
+	require.NoError(err)
+
+	_, err = s.RunStop(context.TODO())
+	require.NoError(err)
+}
+
+func (s *StopTestSuite) TestMissingContainers() {
+	require := s.Require()
+
+	_, err := s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	// start gitbase and bblfsh
+	_, err = s.RunSQL(context.TODO(), "SELECT 1")
+	require.NoError(err)
+
+	// kill the daemon container
+	err = docker.RemoveContainer("srcd-cli-daemon")
+	require.NoError(err)
+
+	// run stop, the other containers should be stopped
+	_, err = s.RunStop(context.TODO())
+	require.NoError(err)
+
+	s.allStopped()
+}
+
+func (s *StopTestSuite) allStopped() {
+	s.T().Helper()
+	require := s.Require()
+
+	out, err := s.RunCommand(context.TODO(), "components", "list")
+	require.NoError(err, out.String())
+
+	expected := regexp.MustCompile(
```

Comment:
is the order guaranteed beforehand?

Reply:
Yes, the list of components is always [sorted here](https://github.com/src-d/engine/blob/9282f346450af6c6bc25884f1c7108b8ed17225f/components/components.go#L220).